kubernetes/resource_kubernetes_secret_v1_data.go

```diff
@@ -0,0 +1,219 @@
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "Secret deleted",
+				Detail:   fmt.Sprintf("The underlying secret %q has been deleted. You should recreate the underlying secret, or remove it from your configuration.", name),
+			}}
+		}
+		return diag.FromErr(err)
+	}
+
+	// Extract managed data from the secret
+	managedData, err := getManagedSecretData(secret)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// filter out the data not managed by terraform
+	configuredData := d.Get("data").(map[string]interface{})
+	for k := range managedData {
+		if _, exists := configuredData[k]; !exists {
+			delete(managedData, k)
+		}
+	}
+	// Update the state with the managed data
+	d.Set("data", managedData)
+	return nil
+}
+
+// extracts data from the secret that is managed by terraform
+func getManagedSecretData(secret *v1.Secret) (map[string]interface{}, error) {
+	managedData := make(map[string]interface{})
+
+	//looping through all data in the secret
+	for key, value := range secret.Data {
+		// decode base64-encoded value
+		decodedValue, err := base64.StdEncoding.DecodeString(string(value))
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode value for key %q: %w", key, err)
+		}
+
+		// just storing the decoded value I got in the managed data map
+		managedData[key] = string(decodedValue)
+	}
+	return managedData, nil
+
+}
+
+func resourceKubernetesSecretV1DataUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	conn, err := m.(KubeClientsets).MainClientset()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	name := metadata.GetName()
+	namespace := metadata.GetNamespace()
+
+	_, err = conn.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if d.Id() == "" {
+			// If we are deleting then there is nothing to do if the resource is gone
+			return nil
+		}
+		if statusErr, ok := err.(*errors.StatusError); ok && errors.IsNotFound(statusErr) {
+			return diag.Errorf("The Secret %q does not exist", name)
+		}
+		return diag.Errorf("Have got the following error while validating the existence of the Secret %q: %v", name, err)
+	}
+	// Craft the patch to update the data
+	data := d.Get("data")
+	if d.Id() == "" {
+		// If we're deleting then we just patch with an empty data map
+		data = map[string]interface{}{}
+	}
+	patchobj := map[string]interface{}{
+		"apiVersion": "v1",
+		"kind":       "Secret",
+		"metadata": map[string]interface{}{
+			"name":      name,
+			"namespace": namespace,
+		},
+		"data": data,
+	}
+	patch := unstructured.Unstructured{}
+	patch.Object = patchobj
+	patchbytes, err := patch.MarshalJSON()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	// Apply the patch
+	_, err = conn.CoreV1().Secrets(namespace).Patch(ctx,
+		name,
+		types.ApplyPatchType,
+		patchbytes,
+		metav1.PatchOptions{
+			FieldManager: d.Get("field_manager").(string),
```

Comment:
The panic from the tests comes the `field_manager` not being included into the schema